src/Shared/CommunicationsUtilities.cs

```diff
@@ -116,6 +116,11 @@ static internal class CommunicationsUtilities
         /// </summary>
         private const int EndOfHandshakeSignal = -0x2a2a2a2a;
 
+        /// <summary>
+        /// The version of the handshake. This should be updated each time the handshake structure is altered.
+        /// </summary>
+        internal const int handshakeVersion = 0x01000000;
```

Comment:
nit: Would be slightly more elegant to define `handshakeVersion = 0x01` and shift left in the `Handshake` ctor. The way it's written now, if you say "bump the version", someone might do `handshakeVersion = 0x01000001`.